src/status.md

```diff
@@ -0,0 +1,313 @@
+
+For now, [the loader](./loader.md) provides the utility necessary to translate between objects in linear memory and JavaScript objects (e.g. with `__newString` and `__getString`), and our hopes are on Type Imports 🦄, Interface Types 🦄 and perhaps Garbage collection 🦄 to make interop more convenient eventually.
+
+See also: [Will interop between AssemblyScript and JavaScript become better?](./frequently-asked-questions.md#will-interop-between-assemblyscript-and-javascript-become-better)
+
+### Union types
+
+WebAssembly cannot efficiently represent locals or globals of dynamic types, so union types are not supported in AssemblyScript. One can however take advantage of the fact that AssemblyScript is a static compiler, with monomorphized generics and [static type checks](./environment.md#static-type-checks), to achieve a similar effect:
+
+```ts
+function addOrConcat<T>(a: T, b: T): T {
+  return a + b; // concats if a string, otherwise adds
+}
+
+function addOrSomethingElse<T>(a: T, b: T): T {
+  if (isString<T>()) {
+    return "something else"; // eliminated if T is not a string
+  } else {
+    return a + b; // eliminated if T is a string
+  }
+}
+```
+
+Another effect of the above is that AssemblyScript does not have an `any` type or `undefined` value.
+
+### Symbols
+
+The standard library implements [`Symbol`](./stdlib/symbol.md), and it is possible to work with and create new symbols, but there is no deep compiler integration like registration of `Symbol.iterator` etc. yet.
+
+### JSON
+
+JSON integration in the compiler itself is still an open question due to its untyped nature. May require a mix of reflection and schema-specific code generation for natural integration.
+
+Solutions being developed by the community:
+
+* [nearprotocol/assemblyscript-json](https://github.com/nearprotocol/assemblyscript-json)
+
+### RegExp
+
+Regular expressions have been on our todo list for quite a while. It's mostly that a good implementation becomes complicated pretty quickly with special Unicode cases, exponential behavior and so on. Also, an ideal RegExp implementation would be compatible with the ECMAScript specification, reasonably fast and integrate deeply with the compiler, so RegExp literals can be pre-compiled (to WebAssembly code or an intermediate bytecode), making it unnecessary most of the time to ship the entire engine with a module.
+
+Solutions being developed by the community:
+
+* [ColinEberhardt/assemblyscript-regex](https://github.com/ColinEberhardt/assemblyscript-regex)
+
+### Closures
+
+Closures (functions with a captured environment) are not yet supported and we are waiting for the Function References 🦄 and Garbage collection 🦄 (captured environments are GC'ed) proposals to land. However, since this is a crucial language feature, we may end up with a filler implementation using linear memory. Not available yet, though.
+
+In the meantime we recommend to restructure code so closures are not necessary, i.e. instead of writing
```

Comment:
Ah! Thank you SOOOOOO much for showing these code examples. Closures itself is kind of a complicated topic. So Showing how to restructure here and what the problem was is amazing! :smile: :tada: